### PR TITLE
Initial commit of endpoint for the scheduling of prisoner court events.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
The is a draft PR as it is work in progress.  After feedback at this point in time.

### CODE REVIEW

File: src/main/java/net/syscon/elite/api/resource/OffenderMovementsResource.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    // TODO - create the model object for POST operation
+    //
+    @PostMapping("/{bookingId}/prison-to-court")
+    @ApiOperation(value = "Schedules a prison to court event for an offender.", notes = "Schedules a prison to court event for an offender.", nickname = "prisonToCourtEvent")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Court event created.", response = CourtEvent.class),
+            @ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "Requested resource not found.", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class)})
+    ResponseEntity<CourtEvent> prisonToCourtEvent(@ApiParam(value = "The offender booking id.", required = true) @PathVariable("bookingId") Long bookingId,
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/resource/OffenderMovementsResource.java]: Think you can use `@ResponseStatus(HttpStatus.CREATED)` as annotation instead

Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/api/resource/OffenderMovementsResource.java]: Done.

File: src/main/java/net/syscon/elite/api/model/PrisonToCourtEvent.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    private Long courtCaseId;
+
+    @ApiModelProperty(required = true, value = "The prison agency code where the offender will be moved from.", position = 2, example = "LEI")
+    private String fromPrisonLocation;
+
+    @ApiModelProperty(required = true, value = "The court agency code where the offender will moved to.", position = 3, example = "LEEDSCC")
+    private String toCourtLocation;
+
+    @ApiModelProperty(required = true, value = "The date and time of the court event.", position = 4, example = "2020-02-28T14:40:00.000Z")
+    private String courtEventDateTime;
```
Comment (mikehalmamoj) [inline src/main/java/net/syscon/elite/api/model/PrisonToCourtEvent.java]: Any reason for not using a LocalDateTime?

File: src/main/java/net/syscon/elite/api/resource/OffenderMovementsResource.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @PostMapping("/{bookingId}/prison-to-court-events")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiOperation(value = "Schedules a prison to court event for an offender.", notes = "Schedules a prison to court event for an offender.", nickname = "prisonToCourtEvent")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Court event created.", response = CourtEvent.class),
+            @ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "Requested resource not found.", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class)})
+    ResponseEntity<CourtEvent> prisonToCourtEvent(@ApiParam(value = "The offender booking to associate the court event with.", required = true) @PathVariable("bookingId") Long bookingId,
+                                                  @ApiParam(value = "The prison to court event to be scheduled for the offender booking.", required = true) @RequestBody PrisonToCourtEvent event);
```
Comment (mikehalmamoj) [inline src/main/java/net/syscon/elite/api/resource/OffenderMovementsResource.java]: Should be final?  (said the pot to the kettle)

File: src/main/java/net/syscon/elite/api/model/PrisonToCourtEvent.java (hunk 2 above)
Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/api/model/PrisonToCourtEvent.java]: Nope. Thanks.

File: src/main/java/net/syscon/elite/api/resource/OffenderMovementsResource.java (hunk 3 above)
Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/api/resource/OffenderMovementsResource.java]: I am kinda on the fence this one.  The 'Impl' is final.

File: src/main/java/net/syscon/elite/api/model/PrisonToCourtEvent.java (hunk 2 above)
Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/api/model/PrisonToCourtEvent.java]: Done

**Pull request merged**